plugins/outputs/postgresql/postgresql.go

```diff
@@ -0,0 +1,424 @@
+  ## connection with the server and doesn't restrict the databases we are trying
+  ## to grab metrics for.
+  ##
+  #connection = "host=localhost user=postgres sslmode=verify-full"
+
+  ## Postgres schema to use.
+  schema = "public"
+
+  ## Store tags as foreign keys in the metrics table. Default is false.
+  tags_as_foreign_keys = false
+
+  ## Suffix to append to table name (measurement name) for the foreign tag table.
+  tag_table_suffix = "_tag"
+
+  ## Deny inserting metrics if the foreign tag can't be inserted.
+  foreign_tag_constraint = false
+
+  ## Store all tags as a JSONB object in a single 'tags' column.
+  tags_as_jsonb = false
+
+  ## Store all fields as a JSONB object in a single 'fields' column.
+  fields_as_jsonb = false
+
+  ## Templated statements to execute when creating a new table.
+  create_templates = [
+    '''CREATE TABLE {{.table}} ({{.columns}})''',
+  ]
+
+  ## Templated statements to execute when adding columns to a table.
+  ## Set to an empty list to disable. Points containing tags for which there is no column will be skipped. Points
+  ## containing fields for which there is no column will have the field omitted.
+  add_column_templates = [
+    '''ALTER TABLE {{.table}} ADD COLUMN IF NOT EXISTS {{.columns|join ", ADD COLUMN IF NOT EXISTS "}}''',
+  ]
+
+  ## Templated statements to execute when creating a new tag table.
+  tag_table_create_templates = [
+    '''CREATE TABLE {{.table}} ({{.columns}}, PRIMARY KEY (tag_id))''',
+  ]
+
+  ## Templated statements to execute when adding columns to a tag table.
+  ## Set to an empty list to disable. Points containing tags for which there is no column will be skipped.
+  tag_table_add_column_templates = [
+    '''ALTER TABLE {{.table}} ADD COLUMN IF NOT EXISTS {{.columns|join ", ADD COLUMN IF NOT EXISTS "}}''',
+  ]
+
+  ## When using pool_max_conns>1, an a temporary error occurs, the query is retried with an incremental backoff. This
+  ## controls the maximum backoff duration.
+  retry_max_backoff = "15s"
+
+  ## Enable & set the log level for the Postgres driver.
+  # log_level = "info" # trace, debug, info, warn, error, none
+`
+
+type Postgresql struct {
+	Connection                 string
+	Schema                     string
+	TagsAsForeignKeys          bool
+	TagTableSuffix             string
+	ForeignTagConstraint       bool
+	TagsAsJsonb                bool
+	FieldsAsJsonb              bool
+	CreateTemplates            []*template.Template
+	AddColumnTemplates         []*template.Template
+	TagTableCreateTemplates    []*template.Template
+	TagTableAddColumnTemplates []*template.Template
+	RetryMaxBackoff            config.Duration
+	LogLevel                   string
+
+	dbContext       context.Context
+	dbContextCancel func()
+	db              *pgxpool.Pool
+	tableManager    *TableManager
+	tagsCache       *freecache.Cache
+
+	writeChan      chan *TableSource
+	writeWaitGroup *utils.WaitGroup
+
+	Logger telegraf.Logger
```

Comment:
Please change to
```suggestion
	Logger telegraf.Logger `toml:"-"`
```
to avoid the logger being set through toml.